molten-http-client/src/main/java/com/hotels/molten/http/client/Protocol.java

```diff
@@ -0,0 +1,48 @@
+package com.hotels.molten.http.client;
+/*
+ * Copyright (c) 2020 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NonNull;
+import reactor.netty.http.HttpProtocol;
+
+/**
+ * The different configurable types of http protocols.
+ */
+@Getter
+@AllArgsConstructor
+public enum Protocol {
+
+    /**
+     * Cleartext HTTP/2 with no "upgrade" round trip.
+     */
+    HTTP_2C(HttpProtocol.H2C, okhttp3.Protocol.H2_PRIOR_KNOWLEDGE),
```

Comment:
I wanted to raise that this adds compile time dependency on reactor-netty and it is an optional dependency but it seems it was changed sometimes in the past. Just a note to @adambriny that we might need to solve this in the future..

Reply:
good point, thanks @danielimre